lib/plugins/aws/metrics/awsMetrics.js

```diff
@@ -0,0 +1,212 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+const chalk = require('chalk');
+const _ = require('lodash');
+const moment = require('moment');
+const validate = require('../lib/validate');
+
+class AwsMetrics {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options;
+    this.provider = this.serverless.getProvider('aws');
+
+    Object.assign(this, validate);
+
+    this.hooks = {
+      'metrics:metrics': () => BbPromise.bind(this)
+        .then(this.extendedValidate)
+        .then(this.getMetrics)
+        .then(this.showMetrics),
+    };
+  }
+
+  extendedValidate() {
+    this.validate();
+
+    // validate function exists in service
+    this.options.function = this.serverless.service.getFunction(this.options.function).name;
```

Comment:
Yes. This call to `getFunction` checks if the function is available in the service.

It's used throughout the codebase.